From: Daniel Golle <daniel@makrotopia.org>
Subject: wwan: Add MeigLink SLM750 modem support

Add patch found in Teltonika RUT9_R_00.07.01.4 GPL SDK download[1]
adding USB IDs of the MeigLink SLM750 to the relevant kernel drivers.
Newer versions of Teltonika's 2G/3G/4G RUT9XX WWAN router series come
with this kind of modem.

[1]: https://wiki.teltonika-networks.com/view/GPL
Submitted-by: Daniel Golle <daniel@makrotopia.org>
---
 drivers/net/usb/qmi_wwan.c                    | 8 ++++++
 1 file changed, 8 insertions(+)

--- a/drivers/net/usb/qmi_wwan.c
+++ b/drivers/net/usb/qmi_wwan.c
@@ -1456,6 +1456,7 @@ static const struct usb_device_id produc
 	{QMI_MATCH_FF_FF_FF(0x2c7c, 0x0620)},	/* Quectel EM160R-GL */
 	{QMI_MATCH_FF_RAWIP_INTF(0x2c7c, 0x0800, sdx55)},	/* Quectel RM500Q-GL */
 	{QMI_MATCH_FF_RAWIP_INTF(0x2c7c, 0x0801, sdx55)},	/* Quectel RM520N-EU */
+	{QMI_MATCH_FF_FF_FF(0x05c6, 0xf601)},   /* MeigLink SLM750 */
 	{QMI_MATCH_FF_FF_FF(0x2dee, 0x4d22)},   /* MeigLink SLM828 */
 
 	{USB_VENDOR_AND_INTERFACE_INFO(0x2dee, 0xff, 0x10, 0x05)}, //rmnet
--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -247,6 +247,8 @@ static void option_instat_callback(struc
 #define UBLOX_PRODUCT_R410M			0x90b2
 /* These Yuga products use Qualcomm's vendor ID */
 #define YUGA_PRODUCT_CLM920_NC5			0x9625
+/* These MeigLink products use Qualcomm's vendor ID */
+#define MEIGLINK_PRODUCT_SLM750			0xf601
 
 #define MEIGLINK_VENDOR_ID 			0x2DEE
 #define MEIG_PRODUCT_SLM790 			0x4D20
@@ -1196,6 +1198,11 @@ static const struct usb_device_id option
 	  .driver_info = ZLP },
 	{ USB_DEVICE(QUECTEL_VENDOR_ID, QUECTEL_PRODUCT_BG96),
 	  .driver_info = RSVD(4) },
+	/* Meiglink products using Qualcomm vendor ID */
+	// Works OK. In case of some issues check macros that are used by Quectel Products
+	{ USB_DEVICE_AND_INTERFACE_INFO(QUALCOMM_VENDOR_ID, MEIGLINK_PRODUCT_SLM750, 0xff, 0xff, 0xff),
+	  .driver_info = NUMEP2 },
+	{ USB_DEVICE_AND_INTERFACE_INFO(QUALCOMM_VENDOR_ID, MEIGLINK_PRODUCT_SLM750, 0xff, 0, 0) },
 	{ USB_DEVICE_AND_INTERFACE_INFO(MEIGLINK_VENDOR_ID, MEIG_PRODUCT_SLM770_NCM, 0xff, 0xff, 0xff),
 	  .driver_info = NUMEP2 },
 	{ USB_DEVICE_AND_INTERFACE_INFO(MEIGLINK_VENDOR_ID, MEIG_PRODUCT_SLM770_NCM, 0xff, 0, 0) },
